### PR TITLE
Remove WithId.

### PR DESCRIPTION
**Pull request by Lasering**
Use the latest openstack-common-clients.

### CODE REVIEW

File: src/main/scala/pt/tecnico/dsi/openstack/nova/services/Quotas.scala [hunk 1]
```diff
@@ -15,28 +15,66 @@ final class Quotas[F[_]: Sync: Client](baseUri: Uri, authToken: Header) extends
...
+   */
+  def applyUsage(projectId: String, userId: Option[String] = None): F[QuotaUsage] =
     super.get(wrappedAt = Some(name), buildUri(projectId, userId) / "detail")
 
   /**
    * Gets default quotas for a project.
    * @param projectId UUID of the project.
    */
-  def getDefaults(projectId: String): F[WithId[Quota]] = super.get(wrappedAt = Some(name), uri / projectId / "defaults")
+  def getDefaults(projectId: String): F[Option[Quota]] =
```
Comment (Lasering) [inline src/main/scala/pt/tecnico/dsi/openstack/nova/services/Quotas.scala:70]: Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-nova-client/pullRequest?prid=5937397)

File: src/main/scala/pt/tecnico/dsi/openstack/nova/services/Quotas.scala [hunk 2]
```diff
@@ -15,28 +15,66 @@ final class Quotas[F[_]: Sync: Client](baseUri: Uri, authToken: Header) extends
 
   private def buildUri(projectId: String, userId: Option[String]): Uri = (uri / projectId).withOptionQueryParam("user_id", userId)
 
+  // In a Rest API if the resource does not exist you return a BadRequest not a NotFound </sarcasm>
+  private def getOption[R: Decoder](uri: Uri): F[Option[R]] = {
```
Comment (Lasering) [inline src/main/scala/pt/tecnico/dsi/openstack/nova/services/Quotas.scala:19]: Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-nova-client/pullRequest?prid=5937397)

File: src/main/scala/pt/tecnico/dsi/openstack/nova/services/Quotas.scala [hunk 3]
```diff
@@ -15,28 +15,66 @@ final class Quotas[F[_]: Sync: Client](baseUri: Uri, authToken: Header) extends
...
+  def apply(projectId: String, userId: Option[String] = None): F[Quota] =
     super.get(wrappedAt = Some(name), buildUri(projectId, userId))
 
   /**
    * Shows quota usage for a project.
    * @param projectId UUID of the project.
    * @param userId id of user to list the quotas for.
    */
-  def getUsage(projectId: String, userId: Option[String] = None): F[WithId[QuotaUsage]] =
+  def getUsage(projectId: String, userId: Option[String] = None): F[Option[QuotaUsage]] =
```
Comment (Lasering) [inline src/main/scala/pt/tecnico/dsi/openstack/nova/services/Quotas.scala:56]: Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-nova-client/pullRequest?prid=5937397)

File: src/test/scala/pt/tecnico/dsi/openstack/nova/QuotasSpec.scala [hunk 4]
```diff
@@ -9,25 +10,62 @@ class QuotasSpec extends Utils {
...
-        quotas.ram shouldBe 50.gibibytes
-        quotas.keyPairs shouldBe 100
-        quotas.serverGroups shouldBe 10
-        quotas.serverGroupMembers shouldBe 10
-        quotas.floatingIps shouldBe 10
-        quotas.instances shouldBe 10
-        quotas.securityGroups shouldBe 10
-        quotas.securityGroupRules shouldBe 20
-      }
+    "get quotas for a project (existing id)" in withStubProject.use[IO, Assertion] { project =>
```
Comment (afonsomatos) [inline src/test/scala/pt/tecnico/dsi/openstack/nova/QuotasSpec.scala:48]: Why are you testing for `_.value.cores`? You're then testing the whole `defaultQuotas`.

**Review by afonsomatos — CHANGES_REQUESTED** [review]
Looks good but the quotas testing needs an explanation.

**Review by afonsomatos — APPROVED** [review]

**Pull request merged**